src/Microsoft.ML.Data/Commands/SaveDataCommand.cs

```diff
@@ -134,7 +134,7 @@ private void RunCore(IChannel ch)
                 var keepColumns = Args.Columns
                     .Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries).ToArray();
                 if (Utils.Size(keepColumns) > 0)
-                    data = ColumnSelectingTransformer.CreateKeep(Host, data, keepColumns);
+                    data = new ColumnSelectingTransformer(Host, keepColumns, null).Transform(data);
```

Comment:
>null [](start = 77, length = 4)

null?

Reply:
The `ColumnSelectingTransformer` takes the `keepColumns` and the `dropColumns`, in this case we only have `keepColumns`.

---
In reply to: [253693146](https://github.com/dotnet/machinelearning/pull/2367#discussion_r253693146) [](ancestors = 253693146)

Reply:
Can we not do that? What is wrong with previous approach?

---
In reply to: [253696143](https://github.com/dotnet/machinelearning/pull/2367#discussion_r253696143) [](ancestors = 253696143,253693146)

Reply:
That method used `IDataTransform` under the hood, so I preferred using the Transformer API.

Reply:
But I am open to hear your suggestions if you think something different is better.

Reply:
So CreateKeep and CreateDrop are helper functions to make it easy to know what operation that we are doing and to prevent any bugs (like putting keep in the drop. I would prefer to keep the use of CreateKeep or CreateDrop

---
In reply to: [253697661](https://github.com/dotnet/machinelearning/pull/2367#discussion_r253697661) [](ancestors = 253697661)

Reply:
Makes sense! Ok, so how about I have CreateDrop and CreateKeep rely on the transformer interface?

Something like:

```csharp
        public static IDataView CreateKeep(IHostEnvironment env, IDataView input, string[] keepColumns, bool keepHidden = false)
            => new ColumnSelectingTransformer(env, keepColumns, null, keepHidden).Transform(input);
```

Reply:
I reverted the changes and made `CreateDrop` and `CreateKeep` internal. I did change their implementation so that they output an `IDataView`, the reason for doing so is that `IDataTransform` is an abstraction that we are trying to avoid in the new API.